File: app/src/main/java/com/esafirm/androidplayground/common/BaseController.java
package com.esafirm.androidplayground.common;

import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bluelinelabs.conductor.Controller;
import com.esafirm.androidplayground.utils.Logger;

public class BaseController extends Controller {

    public BaseController() {
        Logger.clear();
    }

    @NonNull
    @Override
    protected View onCreateView(@NonNull LayoutInflater inflater, @NonNull ViewGroup container) {
        return null;
    }
}

File: app/src/main/java/com/esafirm/androidplayground/common/BaseController.kt
package com.esafirm.androidplayground.common

import com.bluelinelabs.conductor.Controller
import com.esafirm.androidplayground.main.RouterAct
import com.esafirm.androidplayground.utils.ContextProvider
import com.esafirm.androidplayground.libs.Logger

abstract class BaseController : Controller(), ContextProvider {
    init {
        Logger.clear()
    }

    override val requiredContext get() = activity!!

    fun start(maker: () -> Controller) =
        RouterAct.start(requiredContext, ControllerMaker(maker))
}
